src/pages/registration/contributors_tab/components/ContributorCard.tsx

```diff
@@ -114,34 +111,56 @@ const StyledArrowButton = styled(Button)`
   min-width: auto;
 `;
 
-interface AuthorCardProps {
-  author: Contributor;
-  onArrowMove: (direction: 'up' | 'down') => void;
-  onMoveAuthor: (event: React.ChangeEvent<any>) => void;
-  onRemoveAuthorClick: () => void;
-  openContributorModal: (unverifiedAuthor: UnverifiedContributor) => void;
+interface ContributorCardProps {
+  contributor: Contributor;
+  onMoveContributor: (newSequence: number, oldSequence: number) => void;
+  onRemoveContributorClick: () => void;
+  openContributorModal: (unverifiedContributor: UnverifiedContributor) => void;
 }
 
-const AuthorCard = ({
-  author,
-  onArrowMove,
-  onMoveAuthor,
-  onRemoveAuthorClick,
+export const ContributorCard = ({
+  contributor,
+  onMoveContributor,
+  onRemoveContributorClick,
   openContributorModal,
-}: AuthorCardProps) => {
+}: ContributorCardProps) => {
   const { t } = useTranslation('registration');
-  const index = author.sequence - 1;
-  const baseFieldName = `${ContributorFieldNames.CONTRIBUTORS}[${index}]`;
+  const {
+    values: {
+      entityDescription: { contributors },
+    },
+  } = useFormikContext<Registration>();
+
+  const fieldArrayIndex = contributors.findIndex(
+    (c) =>
+      c.identity.id === contributor.identity.id &&
+      c.identity.name === contributor.identity.name &&
+      c.role === contributor.role
```

Comment:
er det nødvendig å gjøre alle disse sjekkene?

Reply:
huff... problemet er at vi ikke har noen god identifikator for contributors, så dette er bare et best effort på å finne indeksen til objektet vi jobber med nå
Er de verifisert så har vi `id`, så den er good. Er de ikke verifisert har vi navn og rolle som vi bare må satse på at er identifiserende nok (men det blir jo et problem hvis det er 2 forfattere med likt navn) 

siden contributors er ett array med ulike roller kan vi heller ikke basere oss på indekser, siden en forfatter som regel vil ha ulik indeks i contributors array og feks forfatter-array. jeg fant i hvert fall ingen bedre løsning 😞

Reply:
skjønner... nei, er vel ikke så mye annet å gjøre. burde vært en id på alle forfattere (og slik som jeg har skjønt det, så blir det det etterhvert som ARP skal byttes ut med Cristin sitt person-api)